src/emit/substrate/target.rs

```diff
@@ -526,15 +532,8 @@ impl<'a> TargetRuntime<'a> for SubstrateTarget {
             .into_int_value();
 
         // set the result
-        let offset = unsafe {
-            binary.builder.build_gep(
-                binary.context.i8_type().array_type(SCRATCH_SIZE),
-                binary.scratch.unwrap().as_pointer_value(),
-                &[i32_zero!(), length],
-                "data_offset",
-            )
-        };
-
+        let offset = create_gep_instruction!(binary, binary.scratch.unwrap(), length);
```

Comment:
The name should be specific to what this function/macro is doing. `create_gep_instruction` is too generic. What you are doing here is fetching a pointer to an offset of the scratch buffer.

Reply:
Okay. I'll amend this and keep it in mind for the remaining of the refactoring